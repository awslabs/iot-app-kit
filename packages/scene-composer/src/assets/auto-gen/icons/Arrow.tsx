import * as React from 'react';
import { SVGProps } from 'react';
const SvgArrow = (props: SVGProps<SVGSVGElement>) => (
  <svg width={150} height={150} viewBox='0 0 41 41' fill='none' xmlns='http://www.w3.org/2000/svg' {...props}>
    <path d='M6.968 20.5h25.095M23.698 29l8.365-8.5-8.365-8.5' stroke='#fff' strokeWidth={4} strokeLinejoin='round' />
  </svg>
);
export default SvgArrow;
