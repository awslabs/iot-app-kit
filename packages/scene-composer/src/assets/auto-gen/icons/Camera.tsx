import * as React from 'react';
import { SVGProps } from 'react';

const SvgCamera = (props: SVGProps<SVGSVGElement>) => (
  <svg data-name='Layer 1' xmlns='http://www.w3.org/2000/svg' width={16} height={16} {...props}>
    <path d='M15.12 10.2v-4l-3 2 3 2zM1 5h10v6H1z' />
  </svg>
);

export default SvgCamera;
