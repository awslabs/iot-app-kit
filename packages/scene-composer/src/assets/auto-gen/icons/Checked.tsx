import * as React from 'react';
import { SVGProps } from 'react';
const SvgChecked = (props: SVGProps<SVGSVGElement>) => (
  <svg width={16} height={12} fill='none' xmlns='http://www.w3.org/2000/svg' {...props}>
    <path d='m1.41 5 4.71 4.71 8.29-8.3' stroke='#545B64' />
  </svg>
);
export default SvgChecked;
