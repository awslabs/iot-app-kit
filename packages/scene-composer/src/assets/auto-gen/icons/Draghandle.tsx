import * as React from 'react';
import { SVGProps } from 'react';

const SvgDraghandle = (props: SVGProps<SVGSVGElement>) => (
  <svg width={4} height={19} fill='none' xmlns='http://www.w3.org/2000/svg' {...props}>
    <path stroke='#879596' d='M.5.5v18M2.961.5v18' />
  </svg>
);

export default SvgDraghandle;
