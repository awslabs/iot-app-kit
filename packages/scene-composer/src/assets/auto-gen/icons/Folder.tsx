import * as React from 'react';
import { SVGProps } from 'react';
const SvgFolder = (props: SVGProps<SVGSVGElement>) => (
  <svg width={14} height={14} fill='none' xmlns='http://www.w3.org/2000/svg' {...props}>
    <path
      fillRule='evenodd'
      clipRule='evenodd'
      d='M12.25 13.125H1.75a.874.874 0 0 1-.875-.875V1.75c0-.484.391-.875.875-.875h4.5c.303 0 .585.157.744.415L8.362 3.5h3.888c.484 0 .875.391.875.875v7.875a.874.874 0 0 1-.875.875Zm-6.487-10.5H2.625v8.75h8.75V5.25h-3.5a.875.875 0 0 1-.745-.415l-1.367-2.21Z'
      fill='#879596'
    />
  </svg>
);
export default SvgFolder;
