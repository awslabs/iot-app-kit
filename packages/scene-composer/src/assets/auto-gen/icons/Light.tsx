import * as React from 'react';
import { SVGProps } from 'react';

const SvgLight = (props: SVGProps<SVGSVGElement>) => (
  <svg width={16} height={17} fill='none' xmlns='http://www.w3.org/2000/svg' {...props}>
    <path d='M8 12.5a4 4 0 1 0 0-8 4 4 0 0 0 0 8Z' fill='currentColor' stroke='currentColor' strokeMiterlimit={10} />
    <path
      d='M4.46 4.97 2.34 2.85M8 3.5v-3M11.531 4.96l2.13-2.12M13 8.5h3M11.54 12.03l2.12 2.13M8 13.5v3M4.472 12.04l-2.12 2.12M3 8.5H0'
      stroke='currentColor'
      strokeMiterlimit={10}
    />
  </svg>
);

export default SvgLight;
