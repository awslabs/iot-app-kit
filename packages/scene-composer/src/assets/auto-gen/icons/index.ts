/** Auto-generated file. DO NOT UPDATE MANUALLY **/
export { default as Arrow } from './Arrow';
export { default as Camera } from './Camera';
export { default as Checked } from './Checked';
export { default as Draghandle } from './Draghandle';
export { default as Folder } from './Folder';
export { default as Light } from './Light';
export { default as Lock } from './Lock';
export { default as Modelref } from './Modelref';
export { default as Orbit } from './Orbit';
export { default as Pan } from './Pan';
export { default as Show } from './Show';
export { default as Tag } from './Tag';
export { default as Unchecked } from './Unchecked';
export { default as Unlock } from './Unlock';
