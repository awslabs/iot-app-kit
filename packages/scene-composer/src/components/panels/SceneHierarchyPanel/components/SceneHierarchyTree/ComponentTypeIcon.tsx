import React from 'react';
import { Icon } from '@cloudscape-design/components';

import { Camera, Light, Modelref, Tag } from '../../../../../assets/auto-gen/icons';
import { KnownComponentType } from '../../../../../interfaces';

const ComponentTypeIcon = ({ type, ...props }: { type: string }) => {
  switch (type) {
    case KnownComponentType.Camera:
      return <Icon svg={<Camera {...props} />} />;
    case KnownComponentType.Light:
      return <Icon svg={<Light {...props} />} />;
    case KnownComponentType.ModelRef:
    case KnownComponentType.SubModelRef:
      return <Icon svg={<Modelref {...props} />} />;
    case KnownComponentType.Tag:
      return <Icon svg={<Tag {...props} />} />;
    default:
      return <></>;
  }
};

ComponentTypeIcon.displayName = ComponentTypeIcon;

export default ComponentTypeIcon;
