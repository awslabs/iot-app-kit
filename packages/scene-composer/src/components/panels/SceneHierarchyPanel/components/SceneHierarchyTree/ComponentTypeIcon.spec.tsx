import renderer from 'react-test-renderer';
import React from 'react';

import { Camera, Light, Modelref, Tag } from '../../../../../assets/auto-gen/icons';

import ComponentTypeIcon from './ComponentTypeIcon';

describe('ComponentTypeIcon', () => {
  [
    ['Camera', { key: 'Camera', icon: Camera }],
    ['Light', { key: 'Light', icon: Light }],
    ['ModelRef', { key: 'ModelRef', icon: Modelref }],
    ['SubModelRef', { key: 'SubModelRef', icon: Modelref }],
    ['Tag', { key: 'Tag', icon: Tag }],
    ['Empty', { key: '', icon: <></> }],
  ].forEach((value) => {
    it(`it should render the ${value[0]} svg`, () => {
      const { key } = value[1] as any;
      const container = renderer.create(ComponentTypeIcon({ type: key }));

      expect(container).toMatchSnapshot();
    });
  });
});
