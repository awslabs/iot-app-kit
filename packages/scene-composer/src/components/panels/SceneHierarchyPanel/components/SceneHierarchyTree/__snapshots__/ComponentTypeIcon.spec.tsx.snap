// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComponentTypeIcon it should render the Camera svg 1`] = `
<div
  data-mocked="Icon"
>
  <div>
    <svg
      data-name="Layer 1"
      viewBox="0 0 16 16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M15.12 10.2v-4l-3 2 3 2zM1 5h10v6H1z"
      />
    </svg>
  </div>
</div>
`;

exports[`ComponentTypeIcon it should render the Empty svg 1`] = `null`;

exports[`ComponentTypeIcon it should render the Light svg 1`] = `
<div
  data-mocked="Icon"
>
  <div>
    <svg
      fill="none"
      height={17}
      width={16}
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M8 12.5a4 4 0 1 0 0-8 4 4 0 0 0 0 8Z"
        fill="currentColor"
        stroke="currentColor"
        strokeMiterlimit={10}
      />
      <path
        d="M4.46 4.97 2.34 2.85M8 3.5v-3M11.531 4.96l2.13-2.12M13 8.5h3M11.54 12.03l2.12 2.13M8 13.5v3M4.472 12.04l-2.12 2.12M3 8.5H0"
        stroke="currentColor"
        strokeMiterlimit={10}
      />
    </svg>
  </div>
</div>
`;

exports[`ComponentTypeIcon it should render the ModelRef svg 1`] = `
<div
  data-mocked="Icon"
>
  <div>
    <svg
      fill="none"
      height={14}
      width={14}
      xmlns="http://www.w3.org/2000/svg"
    >
      <g
        clipPath="url(#modelref_svg__a)"
        stroke="currentColor"
        strokeMiterlimit={10}
      >
        <path
          d="M6.946 13.48.465 10.24V3.758L6.946.518l6.482 3.24v6.482l-6.482 3.24Z"
        />
        <path
          d="M.465 3.76 6.946 7l6.482-3.24M6.945 7v6.482"
        />
      </g>
      <defs>
        <clipPath
          id="modelref_svg__a"
        >
          <path
            d="M0 0h13.889v14H0z"
            fill="#fff"
          />
        </clipPath>
      </defs>
    </svg>
  </div>
</div>
`;

exports[`ComponentTypeIcon it should render the SubModelRef svg 1`] = `
<div
  data-mocked="Icon"
>
  <div>
    <svg
      fill="none"
      height={14}
      width={14}
      xmlns="http://www.w3.org/2000/svg"
    >
      <g
        clipPath="url(#modelref_svg__a)"
        stroke="currentColor"
        strokeMiterlimit={10}
      >
        <path
          d="M6.946 13.48.465 10.24V3.758L6.946.518l6.482 3.24v6.482l-6.482 3.24Z"
        />
        <path
          d="M.465 3.76 6.946 7l6.482-3.24M6.945 7v6.482"
        />
      </g>
      <defs>
        <clipPath
          id="modelref_svg__a"
        >
          <path
            d="M0 0h13.889v14H0z"
            fill="#fff"
          />
        </clipPath>
      </defs>
    </svg>
  </div>
</div>
`;

exports[`ComponentTypeIcon it should render the Tag svg 1`] = `
<div
  data-mocked="Icon"
>
  <div>
    <svg
      fill="none"
      height={14}
      width={14}
      xmlns="http://www.w3.org/2000/svg"
    >
      <g
        stroke="currentColor"
        strokeMiterlimit={10}
      >
        <path
          d="M7 10.5a3.5 3.5 0 1 0 0-7 3.5 3.5 0 0 0 0 7Z"
          fill="currentColor"
        />
        <path
          d="M7 13.125A6.125 6.125 0 1 0 7 .875a6.125 6.125 0 0 0 0 12.25Z"
          strokeWidth={2}
        />
      </g>
    </svg>
  </div>
</div>
`;
